platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/CompassView.java

```diff
@@ -57,105 +57,98 @@ private void initialize(Context context) {
-                setVisibility(View.VISIBLE);
-            }
+        if (enabled && !isHidden()) {
+            resetAnimation();
+            setAlpha(1.0f);
+            setVisibility(View.VISIBLE);
         } else {
-            if (northTimer != null) {
-                northTimer.cancel();
-                northTimer = null;
-            }
-            if (fadeAnimator != null) {
-                fadeAnimator.cancel();
-            }
-            fadeAnimator = null;
+            resetAnimation();
+            setAlpha(0.0f);
             setVisibility(View.INVISIBLE);
         }
     }
 
-    public void update(double direction) {
+    public void update(final double direction) {
         this.direction = direction;
-        setRotation((float) direction);
 
         if (!isEnabled()) {
             return;
         }
 
-        if (direction == 0.0) {
-            if (getVisibility() == View.INVISIBLE) {
+        if (isFacingNorth() && fadeCompassViewFacingNorth) {
+            if (getVisibility() == View.INVISIBLE || fadeAnimator != null) {
                 return;
             }
-
-            if (northTimer == null) {
-                if (fadeAnimator != null) {
-                    fadeAnimator.cancel();
-                }
-                fadeAnimator = null;
-
-                northTimer = new Timer("CompassView North timer");
-                northTimer.schedule(new TimerTask() {
-                    @Override
-                    public void run() {
-                        post(new Runnable() {
-                            @Override
-                            public void run() {
-                                setAlpha(1.0f);
-                                fadeAnimator = ViewCompat.animate(CompassView.this).alpha(0.0f).setDuration(1000).withLayer();
-                                fadeAnimator.setListener(new ViewPropertyAnimatorListenerAdapter() {
-                                    @Override
-                                    public void onAnimationEnd(View view) {
-                                        setVisibility(View.INVISIBLE);
-                                        northTimer = null;
-                                    }
-                                });
-                            }
-                        });
-                    }
-                }, 1000);
-            }
+            postDelayed(this, 500);
+            return;
         } else {
-            if (northTimer != null) {
-                northTimer.cancel();
-                northTimer = null;
-            }
-            if (fadeAnimator != null) {
-                fadeAnimator.cancel();
-            }
+            resetAnimation();
             setAlpha(1.0f);
             setVisibility(View.VISIBLE);
         }
+
+        setRotation((float) direction);
+    }
+
+    public void setCompassFadeFacingNorth(boolean compassFadeFacingNorth) {
+        fadeCompassViewFacingNorth = compassFadeFacingNorth;
+    }
```

Comment:
This is slightly confusing, could we match the name of the setter `setCompassFadeFacingNorth` with the name of the field `fadeCompassViewFacingNorth`?